### ISSUE TITLE
【Released】Iteration plan for June

### ISSUE THREAD

**Issue by SparkSnail** [microsoft/nni#2507]
## Release 1.7
- Release manager: @chicm-ms
- Code freeze date: 7.1
- Demo and retrospective date: 7.3
- Release and next iteration plan date: 7.8
-----
## Planned Items
- [x] use new interface to implement the removed NAS examples @QuanluZhang #2311  6/10 #2541
       TODO: IT and doc 
       +  local IT tf2.0 pipeline @chicm-ms 6/22
- [x] NASUI @ultmaster 
    --- NASUI bugs, UT   6/12 #2552 reviewer: @chicm-ms
- [x] Add Support for NAS Open Benchmarks #2506 @ultmaster   design review done, pr  6/19  , 
- [X] Guidelines to handle large files in codeDir #2364  @ultmaster  book meeting to review pr #2454 
need more changes
- [X] merge parameter utils #2437  TBD @ultmaster  PR #2540 
- [x] WebUI @lvybriage 
    - [x] Enhancement to trial duration display https://github.com/microsoft/nni/issues/1489  PR #2523   being review
    - [x] during and intermediate results pictures update, issue to create issue.   6/5 #2531 
    - [x] Support nested experiment more friendly https://github.com/microsoft/nni/issues/2513  ~6/11~  6/15  pr #2554  and demo reviewer: quanlu,  yuge
    - [x] Hyper-parameter   https://github.com/Microsoft/nni/issues/1483   one week  6/19, pr 6/22  #2586 
reviewers:  @SparkSnail @liuzhe-lz 
    - [x] minior webui  table style issues by @QuanluZhang  6/17  PR #2571 reviewers: quanlu, yuge
- [x] update pai storage, issue #2331  @SparkSnail (PAI 1.01 released)  PR #2536  yuge to review
- [x] AML integration  @SparkSnail ~6/12~ TBD  base on @chi 's training service refactoring.
        6/22 PR 
- [x] register user's tuner as build-in tuner (@chicm-ms )( 5.15 ) ~6/9~ 6/11 for review (#2439  ) , reviewers:  liuzhe, shinai  
- [x] finalize the homepage migration to readthedoc @leckie-chn #2312 (4.24)
- [x] Reuse OpenPAI jobs to run multiple trials  #2521 
- [x] Add OpEvo example #PR 2549   review: @liuzhe-lz 

### Model compression TODOs:
- [x] 1. Model speedup support resnet ( concat, add, conv2d group > 1)  ~6/10~  6/17 @zheng-ningxin    reviewers:  chicm,  quanlu
- [x] 2. Fix bug for model graph with shared nn.Module  #2485 @zheng-ningxin ready for review
- [x] 3. Refactor model pruning framework for automatic pruning (separate weight masking algorithms and pruning/training scheduling algorithms)   PR #2504 update per review comments 6/11 ready
- [x] 4. Support more pruning algorithms  #2490  @suiguoxin update per review comments
- [x] 5. Model sensitivity analysis tool #2435  @zheng-ningxin update per review comments
- [x] lottery ticket pruner, add exporting winning ticket @chicm-ms  PR #2559   ~6/11~  
reviewers:  quanlu, yuge, chenqian 6/22
- [x] flops calculation for model compression and NAS @colorjam 6/8 PR #2535 to schedule a meeting to discuss PR #2535   reviewers:  yuge, quanlu
- [x] mac  downloading timeout   @ultmaster   reviewers:  chicm, liuzhe

## Bugs
- [x] nodejs OOM when `make build` #2475  (@liuzhe-lz )  PR submitted  reviewer: @SparkSnail  @QuanluZhang 
- [x] pipeline failed to install torch 1.5.0+cpu @ultmaster 
- [x] bohb pylint error  @chicm-ms 6/22

backlog:
- [ ] Advanced NAS support (@liuzhe-lz )
- [ ] code coverage @chicm-ms 
- [ ] release pipeline

defer to v1.8
- [ ] TensorFlow v2 NAS performance support improve (@liuzhe-lz )   draft PR #2565,  6/24 ready for review
    - improve `mutable` performance  6/19
- [ ] Fix hyperband and BOHB:   @JCRPaquin     
     *  support various mode:  mixed multiple rounds or strictly staged rounds
         current mixed multi rounds    pending
- [ ] NASUI @ultmaster 
    --- focus mode,  
- [ ]  Fix pytorch import issue for feature engineering  @chicm-ms    pending
- [ ] AMC (automl for model compression) implementation   ~6/17~  how to scale  6/19

- [ ] Fix hyperband and BOHB:   @QuanluZhang  6/23
     *  2 bugs raised by Chenqian 6/22    fix at weekend

- [ ] Search Space of Tuner, quniform  +q or mode q #2397 @QuanluZhang   leave it as is for now
hyperopts:  mod q  (our implementation)
+q:  more user friendly

**Comment by chicm-ms** [comment]
Done
